bugfix/string-agg-removal

fivetran-joemarkiewicz opened the pull request:
## PR Overview
**This PR will address the following Issue/Feature:** Issue #118 

**This PR will result in the following new package version:** `v1.7.0`
<!--- Please add details around your decision for breaking vs non-breaking version upgrade. If this is a breaking change, were backwards-compatible options explored? -->

This will be a breaking change as we are removing fields from an end model and could cause downstream errors with their removal.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**
<!--- Copy/paste the CHANGELOG for this version below. -->

### 🚨 Breaking Changes 🚨: Bug Fixes
- The following fields have been deprecated (removed) as these fields proved to be problematic across warehouses due to the end size of the fields.
  - `errors_since_last_completed_sync`
  - `warnings_since_last_completed_sync`
> Note: If you found these fields to be relevant, you may still reference the error/warning messages from within the underlying `log` table.
- The `fivetran_platform_using_sync_alert_messages` variable has been removed as it is no longer necessary due to the above changes.

### Feature Updates
- The following fields have been added to display the number of error/warning messages sync last completed sync. These fields are intended to substitute the information from deprecated fields listed above.
  - `number_errors_since_last_completed_sync`
  - `number_warnings_since_last_completed_sync`

## PR Checklist
### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt run –full-refresh && dbt test
- [X] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned
- [X] All necessary documentation and version upgrades have been applied
- [X] docs were regenerated (unless this PR does not include any code or yml updates)
- [X] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:
<!--- Provide the steps you took to validate your changes below. -->

Please see the corresponding Height ticket for a link to the Hex document that includes the Hex validations notebook.

### If you had to summarize this PR in an emoji, which would it be?
<!--- For a complete list of markdown compatible emojis check our this git repo (https://gist.github.com/rxaviers/7360908)  --> 
🔢

fivetran-reneeli commented on models/fivetran_platform__connector_status.sql:
```diff
@@ -198,12 +198,9 @@ final as (
-        , {{ fivetran_utils.string_agg("distinct case when connector_recent_logs.event_type = 'WARNING' then connector_recent_logs.message_data else null end", "'\\n'") }} as warnings_since_last_completed_sync
-        {% endif %}
+        coalesce(schema_changes.number_of_schema_changes_last_month, 0) as number_of_schema_changes_last_month,
+        count(distinct case when connector_recent_logs.event_type = 'SEVERE' then connector_recent_logs.message_data else null end) as number_errors_since_last_completed_sync,
```
Will `message_data` always be different across each log record for each connector_id? What happens if there's 2 of the same message_data -- is it intended that they will only be counted once as the same error or warning?

fivetran-joemarkiewicz replied:
Good question, in this case we are just using the `message_data` field to count the warning/error entries. The distinct case is not being applied to the message_data, but instead the event_type. So if two message_data of the exact same verbiage are found in the warning event_type, then both will still be counted as a warning since last sync.

fivetran-joemarkiewicz replied:
Ah yes you're right! I was trying to understand why we used distinct previously, but that was to ensure we do not aggregate the same string over and over again. However, like you mentioned we do in fact want to still count those as error warnings with this new field. I will update to remove the distinct clause in the count statement.

fivetran-joemarkiewicz replied:
Thanks for catching this and continuing to dive into this query! Just updated.

fivetran-reneeli replied:
Gotcha, thanks for digging into why we had it in place before! Looks good!

fivetran-reneeli commented on models/fivetran_platform.yml:
```diff
@@ -53,17 +53,11 @@ models:
-          sync completion. Included by default for non-SQL Server targets, but can be disabled by setting
-          the `fivetran_platform_using_sync_alert_messages` var to False.
+      - name: number_warnings_since_last_completed_sync
+        description: Numeric metric indicating the number of warning messages raised sync the last sync completion.
```
```suggestion
        description: Numeric metric indicating the number of warning messages raised since the last sync completion.
```

fivetran-reneeli commented on models/fivetran_platform.yml:
```diff
@@ -53,17 +53,11 @@ models:
-          Aggregated line-separated list of error messages (in JSON format) raised synce the last 
-          sync completion. Included by default for non-SQL Server targets, but can be disabled by setting
-          the `fivetran_platform_using_sync_alert_messages` var to False.
+      - name: number_errors_since_last_completed_sync
+        description: Numeric metric indicating the number of error messages raised sync the last sync completion.
```
```suggestion
      - name: number_errors_since_last_completed_sync
        description: Numeric metric indicating the number of error messages raised since the last sync completion.
```

fivetran-reneeli commented on README.md:
````diff
@@ -66,7 +66,7 @@ Include the following Fivetran Platform package version range in your `packages.
+    version: [">=1.7.0", "<1.8.0"]
 ```
 
 > Note that altough the source connector is now "Fivetran Platform", the package retains the old name of "fivetran_log".
````
```suggestion
> Note that although the source connector is now "Fivetran Platform", the package retains the old name of "fivetran_log".
```

fivetran-reneeli reviewed (changes requested):
Reviewed with a few suggestions/ question!

fivetran-joemarkiewicz commented:
Thanks @fivetran-reneeli for the quick and thoughtful review! I just addressed your comments and committed your suggestions. Thanks for catching my spelling mistakes 🤦. I also addressed the open question you had and will hold off regenerating the docs until approval.

fivetran-reneeli commented:
Thanks @fivetran-joemarkiewicz , just responded above!

fivetran-joemarkiewicz commented:
@fivetran-reneeli thanks again! I agree with your comment and just applied the changes and regenerated the docs. Let me know if you have any other comments.

fivetran-reneeli reviewed (approved):
lgtm
